src/evidently/collector/app.py

```diff
@@ -107,54 +109,45 @@ async def get_logs(
     return storage.get_logs(id)
 
 
-def check_snapshots_factory(
-    service: CollectorServiceConfig, storage: CollectorStorage
-) -> Callable[[], Awaitable[None]]:
-    async def check_snapshots() -> None:
-        for _, collector in service.collectors.items():
-            if not collector.trigger.is_ready(collector, storage):
-                continue
-            # todo: call async
-            await create_snapshot(collector, storage)
-
-    return check_snapshots
+async def check_snapshots_factory(service: CollectorServiceConfig, storage: CollectorStorage) -> None:
+    for _, collector in service.collectors.items():
+        if not collector.trigger.is_ready(collector, storage):
+            continue
+        await create_snapshot(collector, storage)
 
 
 async def create_snapshot(collector: CollectorConfig, storage: CollectorStorage) -> None:
     async with storage.lock(collector.id):
-        current = storage.get_and_flush(collector.id)
+        current = await run_in_thread_poll_executor(storage.get_and_flush, collector.id)  # FIXME: sync function
```

Comment:
Since this is running in a Litestar application, there's a slight benefit of having this function use the same executor as Litestar does (the executor will be re-used by Litestar), so you could just make use of `litestar.concurrency.sync_to_thread`:

```suggestion
        current = await sync_to_thread(storage.get_and_flush, collector.id)  # FIXME: sync function
```

Goes for the other instances as well.